sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -277,21 +277,22 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
       summary = "")
   }
 
-  def ansiDateTimeError(e: Exception): SparkDateTimeException = {
+  def ansiDateTimeArgumentOutOfRange(e: Exception): SparkDateTimeException = {
```

Comment:
Changes in this PR clash with https://github.com/apache/spark/pull/48242/ by @itholic

Reply:
Let's resolve that ticket first, but keep this one open for now, as I am not sure what complete scope will be of the other ticket and this PRs goal is to remove `_LEGACY_ERROR_TEMP_2000` completely.

Reply:
@MaxGekk I aligned my changes with the @itholic PR, so we can go on and merge this change, then the other PR can improve the message, while this one stays scoped to assigning proper error classes.

Reply:
sgtm